Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,369 @@
+
+
+async def request_cve_data(session: aiohttp.ClientSession, cveid: str) -> dict:
+    """Request for a single CVE data item.
+    """
+    resp = await session.request('GET', f"https://cveawg.mitre.org/api/cve/{cveid}",
+                                 headers={"Accept": "application/json"})
+    return await resp.json()
+
+
+async def collect_cve_versions():
+    """Get all relevant CVE versions, that are relevant for the test according to the severity
+    dictated by the Standard.
+    """
+
+    # CVE fix versions
+    cfvs = list()
+
+    # Request latest version
+    cve_list = requests.get("https://kubernetes.io/docs/reference/issues-security/official-cve-feed/index.json",
+                            headers={"Accept": "application/json"}).json()
+
+    async with aiohttp.ClientSession() as session:
+        tasks = []
+        for cve in cve_list['items']:
+            cveid = cve['external_url'].split("=")[-1]
+            tasks.append(request_cve_data(session=session, cveid=cveid))
+        # asyncio.gather() will wait on the entire task set to be
+        # completed.  If you want to process results greedily as they come in,
+        # loop over asyncio.as_completed()
+        cve_data_list = await asyncio.gather(*tasks, return_exceptions=True)
+
+    for cve_data in cve_data_list:
+        try:
+            cve_cna = cve_data['containers']['cna']
+            cve_metrics = cve_cna['metrics']
+            # This data is extracted like this due to the location of the relevant information in the schema.
+            # mitre.org uses CVE schema 5.0, which is described in the link below
+            # https://github.com/CVEProject/cve-schema/tree/master/schema/v5.0
+            # The containers -> cna path contains vulnerability information like severity, which is documented
+            # under the metrics list.
+        except KeyError as e:
+            logger.debug(f"They key {e} couldn't be found in the CVE json data for CVE {cveid}.")
+            continue
+        for i, cve_metric in enumerate(cve_metrics):
+            try:
+                cve_affected_versions = cve_cna['affected'][i]['versions']
+            except KeyError as e:
+                logger.debug(f"They key {e} couldn't be found in the CVE json data for CVE {cveid}.")
+                continue
+
+            for key in cve_metric:
+                if re.search(r'[cC][vV][sS]{1,2}V\d', key) and \
+                   cve_cna['affected'][i]['product'] == "Kubernetes" and \
+                   cve_metric[key]['baseScore'] >= CVE_SEVERITY:
+
+                    for v in cve_affected_versions:
+                        vi = K8sVersionInfo()
+                        if 'lessThanOrEqual' in v:
+                            vi = vi.extract_version(v['lessThanOrEqual'].strip("v"))
+                            vi.patch += 1
+                        elif 'lessThan' in v:
+                            vi = vi.extract_version(v['lessThan'].strip("v"))
```

Comment:
The `classmethod` should be called as such:

```suggestion
                        if 'lessThanOrEqual' in v:
                            vi = K8sVersionInfo.extract_version(v['lessThanOrEqual'].strip("v"))
                            vi.patch += 1
                        elif 'lessThan' in v:
                            vi = K8sVersionInfo.extract_version(v['lessThan'].strip("v"))
                        else:
                            vi = K8sVersionInfo()  # not sure what this will do for us actually
```

Reply:
Changed that up.